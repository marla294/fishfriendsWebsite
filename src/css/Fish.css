.fish {
	width: 100%;
	display: grid;
	grid-template-columns: auto 1fr;
	border-radius: 1em;
	box-sizing: border-box;
	border: 0.08em solid var(--blue);
}

.fish:hover {
	color: var(--yellow);
	cursor: pointer;
}

.fishInner {
	display: grid;
	grid-template-columns: auto 1fr;
	grid-gap: 0.5em;
	width: 100%;
}

.selected {
	color: var(--yellow);
	border: 0.08em solid var(--yellow);
	box-sizing: border-box;
}

.compatible-yes {
	background-color: var(--green);
	border: 0.08em solid var(--green);
}

.compatible-maybe {
	background-color: var(--yellow);
	border: 0.08em solid var(--yellow);
}

.compatible-no {
	background-color: var(--red);
	border: 0.08em solid var(--red);
}

.compatible-yes:hover, .compatible-maybe:hover, .compatible-no:hover {
	color: darkblue;
}

.selected:hover {
	color: var(--yellow);
}

.fish img {
	width: 2em;
	margin: 0.2em;
}

.fish .name {
	align-self: center;
	justify-self: left;
}

.small {
	height: 2.6em;
	font-size: 0.75em;
	grid-gap: 0.1em;
}