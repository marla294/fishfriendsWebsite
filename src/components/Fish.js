import React from "react";
import PropTypes from "prop-types";
import "../css/Fish.css";

class Fish extends React.Component {
	getImageSource = () => {
		let pictureName = this.props.name;

		if (pictureName === "corals/clams") {
			pictureName = "corals and clams";
		}

		if (pictureName === "crabs, shrimps and snails") {
			pictureName = "crabs shrimp and snails";
		}

		return `/images/fish/${pictureName}.png`;
	};

	render() {
		return (
			<div className="fish">
				<img src={this.getImageSource()} alt="fishPic" />
				<div className="name">{this.props.name}</div>
			</div>
		);
	}
}

Fish.propTypes = {
	name: PropTypes.string.isRequired
};

export default Fish;
