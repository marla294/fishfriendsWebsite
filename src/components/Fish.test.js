import React from "react";
import Fish from "./Fish";
import renderer from "react-test-renderer";

test("Correct fish name appears", () => {
	const component = renderer.create(<Fish name="Gobies" />);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

