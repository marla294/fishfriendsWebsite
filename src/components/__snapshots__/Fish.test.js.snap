// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct fish name appears 1`] = `
<div
  className="fish"
>
  <img
    alt="fishPic"
    src="/images/fish/Gobies.png"
  />
  <div
    className="name"
  >
    Gobies
  </div>
</div>
`;
